public/app/features/alerting/unified/components/mute-timings/MuteTimingTimeInterval.tsx

```diff
@@ -146,21 +157,94 @@ export const MuteTimingTimeInterval = () => {
   );
 };
 
+interface DaysOfTheWeekProps {
+  defaultValue?: string;
+  onChange: (input: string) => void;
+}
+
+const parseDays = (input: string): string[] => {
+  const parsedDays = input
+    .split(',')
+    .map((day) => day.trim())
+    // each "day" could still be a range of days, so we parse the range
+    .flatMap((day) => (day.includes(':') ? parseWeekdayRange(day) : day))
+    .map((day) => day.toLowerCase())
+    // remove invalid weekdays
+    .filter((day) => DAYS_OF_THE_WEEK.includes(day));
+
+  return uniq(parsedDays);
+};
+
+// parse monday:wednesday to ["monday", "tuesday", "wednesday"]
+function parseWeekdayRange(input: string): string[] {
+  const [start = '', end = ''] = input.split(':');
+
+  const startIndex = DAYS_OF_THE_WEEK.indexOf(start);
+  const endIndex = DAYS_OF_THE_WEEK.indexOf(end);
+
+  return DAYS_OF_THE_WEEK.slice(startIndex, endIndex + 1);
+}
+
+const DaysOfTheWeek = ({ defaultValue = '', onChange }: DaysOfTheWeekProps) => {
```

Comment:
These changes are for parsing the Alertmanager config for weekdays in to our visual weekday selector. 

Valid config is for example `monday, wednesday:friday sunday` which translates to `monday, wednesday, thursday, friday, sunday`